application-accelerator/creating-accelerators/custom-types.hbs.md

```diff
@@ -0,0 +1,163 @@
+# Using Custom Types
+Application Accelerators can declare new types to be used for options declaration,
+in addition to the built-in types `string`, `number` and `boolean`.
+
+## Introducing the Types Section
+
+The `accelerator.yaml` file syntax supports an entry named `types` inside the top-level
+`accelerator` section, allowing the definition of custom types, with support for
+`struct` types, like so:
+```yaml
+accelerator:
+  options:
+    ...
+  types:
+    - name: Task
+      struct:
+        - name: title
+          dataType: string
+          label: Title
+          description: A sample title
+        - name: details
+          label: Task details
+          description: Enter the task details
+        - name: done
+          dataType: boolean
+          label: Done?
+          defaultValue: false
+```
+
+The `struct` type definition is syntactically equivalent to **a sequence (hence ordered) of
+option definitions**.
+
+A type name MUST be a `CapitalizedName`.
+
+## Interaction with UIs and Values Submission
+
+The effect of the previous example is that a new _type_ then becomes available
+(in addition to the built-in `string`, `number`, `boolean` and sequences thereof)
+for the `dataType` property of any option:
+```yaml
+accelerator:
+  options:
+    - name: myTask
+      dataType: Task
+  types:
+    ...
+```
+whereby UIs would render something similar to the following:
+![SimpleTask](simple-task.png)
+
+and associate the entered
```

Comment:
Is this a run on sentence from somewhere - can this be reworded - it's not very clear.

Reply:
Yes it  is (it was, I saw you updated it).
Sadly the sentence doesn't make sense anymore, sorry.

There was a train of thoughts in the previous sentence.
It BECAUSE you'd use such yaml construct THAT uis would render such widget AND THAT the entered values would be associated with the example value submission. You've changed the subject of  the sentence (and lost the first BECAUSE/THEN structure).